### ISSUE TITLE
About inference speed

### ISSUE THREAD

**Issue by YingxuanW** [noahzn/Lite-Mono#174]
Hi! Sorry to bother you again! Could you please tell me about how you do experiments on inference speed? Is there any code for that? I tried to add some codes on eval.py but it did not work well.

**Comment by noahzn** [comment]
Hi, what do you mean "it didn't work well". Please check https://github.com/noahzn/Lite-Mono/blob/4874b35df8ed4da16159ce8be8c697028b72bf76/evaluate_depth.py#L130C16-L132C33

**Comment by YingxuanW** [comment]
`total_time_ms = 0.0
        total_images = 0
        with torch.no_grad():
            for data in dataloader:
                input_color = data[("color", 0, 0)].cuda()
                orig_batch_size = input_color.size(0)
                if opt.post_process:
                    # Post-processed results require each image to have two forward passes
                    input_color = torch.cat((input_color, torch.flip(input_color, [3])), 0)

                flops, params, flops_e, params_e, flops_d, params_d = profile_once(encoder, depth_decoder, input_color)
                t1 = time_sync()
                output = depth_decoder(encoder(input_color))
                t2 = time_sync()

                # 在模型推理之后计算并打印推理时间
                batch_time_ms = (t2 - t1) * 1000  # 转换为毫秒
                total_time_ms += batch_time_ms
                total_images += orig_batch_size

                pred_disp, _ = disp_to_depth(output[("disp", 0)], opt.min_depth, opt.max_depth)
                pred_disp = pred_disp.cpu()[:, 0].numpy()

                if opt.post_process:
                    N = pred_disp.shape[0] // 2
                    pred_disp = batch_post_process_disparity(pred_disp[:N], pred_disp[N:, :, ::-1])

                pred_disps.append(pred_disp)

        avg_time_per_image = total_time_ms / total_images
        print("\nOverall average inference time per image: {:.3f} ms".format(avg_time_per_image))
        print(f"平均推理时间：{avg_time_per_image} = 总推理时间/ 总图片数：{total_time_ms}/{total_images}")

        pred_disps = np.concatenate(pred_disps)
`
I test the inference speed of the model without pre-training:
Overall average inference time per image: 5.454 ms
平均推理时间：5.454215291924935 = 总推理时间/ 总图片数：3801.5880584716797/697

**Comment by YingxuanW** [comment]
But I remember it should be 4.5ms

**Comment by noahzn** [comment]
Please only count the time of inference from the 100th image. You need `warmup` to have a more accurate time evaluation.

**Comment by YingxuanW** [comment]
Aha! 
Overall average inference time per image: 4.462 ms!
